server/src/main/java/org/elasticsearch/gateway/LocalAllocateDangledIndices.java

```diff
@@ -52,24 +53,24 @@ public class LocalAllocateDangledIndices {
     public static final String ACTION_NAME = "internal:gateway/local/allocate_dangled";
 
     private final TransportService transportService;
-
     private final ClusterService clusterService;
-
     private final AllocationService allocationService;
-
     private final IndexMetadataVerifier indexMetadataVerifier;
+    private final ThreadPool threadPool;
 
     @Inject
     public LocalAllocateDangledIndices(
         TransportService transportService,
         ClusterService clusterService,
         AllocationService allocationService,
-        IndexMetadataVerifier indexMetadataVerifier
+        IndexMetadataVerifier indexMetadataVerifier,
+        ThreadPool threadPool
     ) {
         this.transportService = transportService;
         this.clusterService = clusterService;
         this.allocationService = allocationService;
         this.indexMetadataVerifier = indexMetadataVerifier;
+        this.threadPool = threadPool;
```

Comment:
I think we could just use `transportService.getThreadPool()` (or `clusterService.getClusterApplierService().threadPool()`), no need to capture this separately.

Reply:
I did this initially but then decided to wire own thread pool instead of "borrowing" it from somewhere else. I am concerned about relying on implementation details of other services

Reply:
If we can't rely on `transportService.getThreadPool()` then we have bigger concerns :)